src/runtime_src/core/common/config_reader.h

```diff
@@ -289,6 +289,29 @@ get_profile_api()
   return value;
 }
 
+inline bool
+get_host_trace()
+{
+  static bool value = detail::get_bool_value("Debug.host_trace", false);
+  return value;
+}
+
+inline bool
+get_host_trace_once()
+{
+  static bool called = false;
+
+  // host_trace is a generic switch that should enable one layer of 
+  //  trace (either OpenCL level, Native XRT API level, or HAL level).
+  //  Whichever function is called from the host code first should enable
+  //  that level of trace.  Therefore, an OpenCL application will only
+  //  enable OpenCL trace, a Native XRT API application will only enable
+  //  Native XRT trace, and a HAL level application will only enable HAL trace.
+  bool result = get_host_trace() && !called;
+  called = true ;
+  return result ;
```

Comment:
This is not thread safe.  `called` can be written by one thread on line 311, while some other thread computes 310 and a third read executes 312.    That sort of logic does not belong here in this header I won't even claim to understand what it does and why, the comment didn't help me.

Reply:
The function was renamed and moved out of the config_reader.h file into the utils.cpp file, multithreaded protection was added via lock_guard, and the comments were redone to clarify what host_trace=true is meant to enable and what the load_host_trace() functionality and usage is.